main.py

```diff
@@ -1,116 +1,99 @@
-                        # Return as cwd and recursively check if git repo
-                        GetAllGitReposCwd(fullpath)
+def NOOP(**_):
+    pass
 
-    # Call Function & Build List of Paths 2 Git Repos
-    GetAllGitReposCwd()
 
+def commit_changes(start_directory, commit_msg=None):
+    git_commands = []
+    logging.debug("Start dir '%s'. Msg '%s'.", start_directory, commit_msg)
+    for directory in [d for d in listdir(start_directory) if path.isdir(path.join(start_directory, d, '.git'))]:
+        logging.debug("Found a git dir '%s'", directory)
+        git_dir = path.join(start_directory, directory)
+        base_cmd = ['git', '-C', git_dir]
 
-    # Logging
-    print('\n The following git repositories have been found => {}'.format(git_repos))
+        cmd = []
+        cmd.extend(base_cmd)
+        cmd.append('status')
+        git_status = subprocess.check_output(cmd)
 
+        if ALL_CLEAR_MSG not in str(git_status):
+            if not commit_msg:
+                cmd_commit_msg = input(f"Please enter a commit message for '{directory}': ")
+                cmd_commit_msg = cmd_commit_msg if cmd_commit_msg else COMMIT_MSG
+            else:
+                cmd_commit_msg = commit_msg
 
+            cmd = []
+            cmd.extend(base_cmd)
+            cmd.extend(['add', '--all'])
+            git_commands.append(cmd)
 
-    # Step 2:  Check Status of All Git Repos _________________________________
+            cmd = []
+            cmd.extend(base_cmd)
+            cmd.extend(['commit', '-m', f'\'{cmd_commit_msg}\''])
+            git_commands.append(cmd)
 
-    # Check if list is empty
-    if not git_repos:
-        print('No git repositories found in cwd directory tree.  Ending process')
+    return git_commands
 
-    # Otherwise, proceed
-    else:
-        # Iterate List of Repository Paths
-        for repo in git_repos:
-            # Change CWD to Repo Path
-            os.chdir(repo)
-            print('\n***************************************************')
 
-            # Obj to catch git status msg
-            git_status_msg = ''
-            msg = "modified"
+def command_handler(args):
+    git_commands = commit_changes(getattr(args, 'dir', '.'), getattr(args, 'msg', None))
 
-            # Redirect Stdout
-            cmd = ['git', 'status']
-            git_output = subprocess.check_output(cmd)
+    for cmd in git_commands:
+        if args.dry:
+            print(' '.join(cmd))
+        else:
+            subprocess.check_output(cmd)
 
-            # Check if Repository Is Not up-to-date
-            if msg in str(git_output.decode('utf-8')):
+    if args.dry:
+        print('No commands have been executed during this process.')
 
-                # Logging
-                print('Repository => {} is not up to date.  Returning git status\n'.format(repo))
-                print(git_output.decode('utf-8'))
 
-                # Ask user If he/she wants to stage changes
```

Comment:
That clears the next line up. I really wasn't sure what was going on.

Great documentation.